code/workspaces/infrastructure-api/src/controllers/stacksController.js

```diff
@@ -24,28 +29,43 @@ function listByMount(request, response) {
   return controllerHelper.validateAndExecute(request, response, errorMessage, listByMountExec);
 }
 
+function listByProjectExec(request, response) {
+  const { user } = request;
+  const params = matchedData(request);
+  return stackRepository.getAllByProject(params.projectKey, user)
+    .then(mapHandleId)
```

Comment:
Probably not worth changing but when I was adding the project endpoints @gary-lloyd-tessella mentioned that this ID handling can be passed off to the GraphQL resolvers quite nicely. For example, the `Project` resolver is as follows. This means it only needs to be defined in that single place rather than everywhere you get an item from Mongo.
```js
Project: {
  id: obj => (obj._id), // eslint-disable-line no-underscore-dangle
  projectUsers: (obj, args, ctx) => userService.getProjectUsers(obj.key, ctx.token),
  accessible: (obj, args, ctx) => userService.isMemberOfProject(obj.key, ctx.token),
},
```